### PR TITLE
Fix #719 content option rules

### LINKED ISSUE #719: $content modifier works as URL exclusion
This modifier must be ignored by extension.
image

### PR DESCRIPTION
**Pull request by Mizzick**
#719

### CODE REVIEW

File: Extension/lib/filter/rules/url-filter-rule.js [hunk 1]
```diff
@@ -560,7 +560,11 @@
                     } else if (optionName[0] === api.FilterRule.NOT_MARK && optionName.substring(1) in UrlFilterRule.contentTypes) {
                         restrictedContentType |= UrlFilterRule.contentTypes[optionName.substring(1)]; // jshint ignore:line
                     } else if (optionName in UrlFilterRule.ignoreOptions) { // jshint ignore:line
-                        // Ignore
+                        if (optionName === 'CONTENT' && optionsParts.length === 1) {
```
Comment (atropnikov) [inline Extension/lib/filter/rules/url-filter-rule.js]: What about `test.com$content,domain=example.com`?

Comment (Mizzick) [inline Extension/lib/filter/rules/url-filter-rule.js]: > A rule with multiple modifiers which includes $content should continue to work.

Comment (atropnikov) [inline Extension/lib/filter/rules/url-filter-rule.js]: It means "content-type" modifiers.

Comment (Mizzick) [inline Extension/lib/filter/rules/url-filter-rule.js]: ok!

File: Extension/lib/filter/rules/url-filter-rule.js [hunk 2]
```diff
@@ -573,6 +576,11 @@
         if (restrictedContentType > 0) {
             this.restrictedContentType = restrictedContentType;
         }
+
+        if (this.contentOption && this.permittedContentType === UrlFilterRule.contentTypes.ALL) {
```
Comment (atropnikov) [inline Extension/lib/filter/rules/url-filter-rule.js]: Make it simpler, without a new boolean variable.
```
if( this.permittedContentType === CONTENT || this.restrictedContentType === CONTENT) {
   throw ...
}
```

Comment (Mizzick) [inline Extension/lib/filter/rules/url-filter-rule.js]: ok!

File: Extension/lib/filter/rules/url-filter-rule.js [hunk 3]
```diff
@@ -554,7 +554,7 @@
                     }
                     break;
                 case UrlFilterRule.CONTENT_OPTION:
```
Comment (atropnikov) [inline Extension/lib/filter/rules/url-filter-rule.js]: You don't need this 'case') It will be processed in the `default` case.

Comment (Mizzick) [inline Extension/lib/filter/rules/url-filter-rule.js]: okaay!

File: Extension/lib/filter/rules/url-filter-rule.js (hunk 1 above)
Comment (ameshkov) [inline Extension/lib/filter/rules/url-filter-rule.js]: @atropnikov frankly, I agree with @Mizzick, and I like his implementation much more than what we have at the moment.

`$content` is a document-level exception modifier, just like `$elemhide`. There're no rules with `$content` and `$domain`  at the same time.

**Pull request merged**